TeamCode/src/main/java/org/firstinspires/ftc/teamcode/ImplConceptAlgoMagnitude.java

```diff
@@ -0,0 +1,129 @@
+ * promote products derived from this software without specific prior written permission.
+ *
+ * NO EXPRESS OR IMPLIED LICENSES TO ANY PARTY'S PATENT RIGHTS ARE GRANTED BY THIS
+ * LICENSE. THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.firstinspires.ftc.teamcode;
+
+import com.qualcomm.robotcore.eventloop.opmode.OpMode;
+import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
+import com.qualcomm.robotcore.util.ElapsedTime;
+
+/**
+ * algorithm for calculation of magnitude
+ * for use with advanced mecanum drives
+ * jlemke 9/26/2017 @ 7:26 AM
+ */
+
+@TeleOp(name="Mecanum Magnitude Algorithm fConcept", group="Concept")
+public class ImplConceptAlgoMagnitude extends OpMode
+{
+    // move that gear up
+    Hardware750 robot = new Hardware750();
+    // setup runtime timer
+    private ElapsedTime runtime = new ElapsedTime();
+
+    @Override
+    public void init() {
+        telemetry.addData("Status", "Uninitialized...");
+        robot.init(hardwareMap);
+        telemetry.addData("Status", "Initialized");
+    }
+
+    /*
+     * Code to run REPEATEDLY after the driver hits INIT, but before they hit PLAY
+     */
+    @Override
+    public void init_loop() {
+    }
+
+    @Override
+    public void start() {
+        runtime.reset();
+    }
+
+    @Override
+    public void loop() {
+        telemetry.addData("LeftTrig", gamepad1.left_trigger);
+        telemetry.addData("RightTrig", gamepad1.right_trigger);
+        telemetry.addData("lstick X", gamepad1.left_stick_x);
+        telemetry.addData("lstick Y", (-1 * gamepad1.left_stick_y));
+        double presquare = Math.pow(gamepad1.left_stick_x, 2) + Math.pow(gamepad1.left_stick_y, 2);
+        telemetry.addData("presquare", presquare);
+        double magnitude = ((-1 * gamepad1.left_stick_y) < 0) ? -1*Math.sqrt(presquare) : Math.sqrt(presquare);
+        if (magnitude > 1) {
+            magnitude = 1;
+        }
+        telemetry.addData("magnitude", magnitude);
+        double angle = Math.atan2((-1*gamepad1.left_stick_y),gamepad1.left_stick_x);
+        if (angle < 0) {
+            angle += Math.PI*2;
+        }
+        double leftTrig = (-1 * gamepad1.left_trigger);
+        double rightTrig = gamepad1.right_trigger;
```

Comment:
What I recommend you do with this (and the line above it) is actually sum them together to produce a single value (our algorithms expect a value between `[-1, 1]`) This can be done pretty simply with something like
```java
double triggerVal = (-1 * gamepad1.left_trigger) + gamepad1.right_trigger;
```
The way it is right now, you won't be subtracting or adding positive or negative numbers when you should.